[gateway-gcs] Setting bucket policy to public returns an odd error response.

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
Either should succeed or fail properly with "NotImplemented".

<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->

## Current Behavior
```
PUT /minio-go-testdg1x033jhmlgnozvf/?policy= HTTP/1.1
Host: localhost:9000
User-Agent: Minio (linux; amd64) minio-go/2.1.0 Minio-go-FunctionalTest/0.1.0
Content-Length: 463
Authorization: AWS4-HMAC-SHA256 Credential=minio/20170511/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=0d3bf646885cc334ea3a35f705f8c40ebab1242ce937299690c6152a0f67df73
Content-Md5: aQKPZdxkE9hdjw0D5Gn66Q==
X-Amz-Content-Sha256: 66cc03a40c986f6ef7fe4837c4dbce2eae3b94b9b83d3296b6ce4b7fd1589481
X-Amz-Date: 20170511T080652Z
Accept-Encoding: gzip

HTTP/1.1 000 000 status code 000
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: application/xml
Date: Thu, 11 May 2017 08:06:52 GMT
Server: Minio/DEVELOPMENT.2017-05-11T06-52-24Z (linux; amd64)
Vary: Origin
X-Amz-Bucket-Region: us-east-1
X-Amz-Request-Id: 14BD7EBF9A0C6189

e1
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code></Code><Message></Message><Key></Key><BucketName></BucketName><Resource>/minio-go-testdg1x033jhmlgnozvf/</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0
---------END-HTTP---------
--- FAIL: TestFunctional (4.83s)
	api_functional_v4_test.go:2196: Error: Error response code .
FAIL
exit status 1
FAIL	github.com/minio/minio-go	6.152s
```

<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

## Possible Solution
Not sure.
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

## Steps to Reproduce (for bugs)
Just run `minio-go` functional tests.

```diff
PUT /minio-go-testdg1x033jhmlgnozvf/?policy= HTTP/1.1
Host: localhost:9000
User-Agent: Minio (linux; amd64) minio-go/2.1.0 Minio-go-FunctionalTest/0.1.0
Content-Length: 463
Authorization: AWS4-HMAC-SHA256 Credential=minio/20170511/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=0d3bf646885cc334ea3a35f705f8c40ebab1242ce937299690c6152a0f67df73
Content-Md5: aQKPZdxkE9hdjw0D5Gn66Q==
X-Amz-Content-Sha256: 66cc03a40c986f6ef7fe4837c4dbce2eae3b94b9b83d3296b6ce4b7fd1589481
X-Amz-Date: 20170511T080652Z
Accept-Encoding: gzip

HTTP/1.1 000 000 status code 000
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: application/xml
Date: Thu, 11 May 2017 08:06:52 GMT
Server: Minio/DEVELOPMENT.2017-05-11T06-52-24Z (linux; amd64)
Vary: Origin
X-Amz-Bucket-Region: us-east-1
X-Amz-Request-Id: 14BD7EBF9A0C6189

e1
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code></Code><Message></Message><Key></Key><BucketName></BucketName><Resource>/minio-go-testdg1x033jhmlgnozvf/</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0
---------END-HTTP---------
--- FAIL: TestFunctional (4.83s)
	api_functional_v4_test.go:2196: Error: Error response code .
FAIL
exit status 1
FAIL	github.com/minio/minio-go	6.152s
```

```
go test -run TestFunctional -v
```

## Context
GCS Testing.

<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

## Your Environment
```
Version: 2017-05-11T06:52:24Z
Release-Tag: DEVELOPMENT.2017-05-11T06-52-24Z
Commit-ID: 8dcd491ff6255ebaf276dd412114b2e4ddc7af77
```

--- a/cmd/gateway-gcs-anonymous.go
+++ b/cmd/gateway-gcs-anonymous.go
@@ -70,7 +70,6 @@ func (l *gcsGateway) AnonGetObjectInfo(bucket string, object string) (objInfo Ob
 	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
-		fmt.Println(resp.StatusCode)
 		return objInfo, gcsToObjectError(traceError(anonErrToObjectErr(resp.StatusCode, bucket, object)), bucket, object)
 	}
 

--- a/cmd/gateway-gcs-layer.go
+++ b/cmd/gateway-gcs-layer.go
@@ -798,27 +798,21 @@ func (l *gcsGateway) SetBucketPolicies(bucket string, policyInfo policy.BucketAc
 	}
 
 	acl := l.client.Bucket(bucket).ACL()
-
 	if policies[0].Policy == policy.BucketPolicyNone {
 		if err := acl.Delete(l.ctx, storage.AllUsers); err != nil {
 			return gcsToObjectError(traceError(err), bucket)
 		}
-
 		return nil
 	}
 
-	role := storage.RoleReader
-
+	var role storage.ACLRole
 	switch policies[0].Policy {
 	case policy.BucketPolicyReadOnly:
 		role = storage.RoleReader
 	case policy.BucketPolicyWriteOnly:
 		role = storage.RoleWriter
-	case policy.BucketPolicyReadWrite:
-		// not supported, google only has owner role
-		return gcsToObjectError(traceError(NotSupported{}), bucket)
 	default:
-		return gcsToObjectError(traceError(fmt.Errorf("Unknown policy: %s", policies[0].Policy)), bucket)
+		return traceError(NotImplemented{})
 	}
 
 	if err := acl.Set(l.ctx, storage.AllUsers, role); err != nil {
@@ -840,13 +834,10 @@ func (l *gcsGateway) GetBucketPolicies(bucket string) (policy.BucketAccessPolicy
 	policyInfo := policy.BucketAccessPolicy{Version: "2012-10-17"}
 
 	for _, r := range rules {
-		if r.Entity != storage.AllUsers {
+		if r.Entity != storage.AllUsers || r.Role == storage.RoleOwner {
 			continue
 		}
-
 		switch r.Role {
-		case storage.RoleOwner:
-			return policy.BucketAccessPolicy{}, gcsToObjectError(traceError(NotSupported{}), bucket)
 		case storage.RoleReader:
 			policyInfo.Statements = policy.SetPolicy(policyInfo.Statements, policy.BucketPolicyReadOnly, bucket, "")
 		case storage.RoleWriter:
@@ -861,7 +852,7 @@ func (l *gcsGateway) GetBucketPolicies(bucket string) (policy.BucketAccessPolicy
 func (l *gcsGateway) DeleteBucketPolicies(bucket string) error {
 	acl := l.client.Bucket(bucket).ACL()
 
-	// this only removes the storage.AllUsers policies
+	// This only removes the storage.AllUsers policies
 	if err := acl.Delete(l.ctx, storage.AllUsers); err != nil {
 		return gcsToObjectError(traceError(err), bucket)
 	}

